### ISSUE TITLE
support `run --env` flag

### ISSUE THREAD

**Issue by motdotla** [dotenvx/dotenvx#92]
sometimes you just want to set one or two environment variables inline rather than create a .env file. add support for that with the `--env, -e` flag

**Comment by feledori** [comment]
Don't know if this would be possible but prefixing scripts in package.json can clutter the command.
Ideal would be to have one env setup command and combine this with ```&&```. I know that this spawns two seperate processes so everything after && won't know about anything from the command before. This is just an idea but if that would be possible it would improve DX a ton.

Example:
From this:
```json
{
    "scripts": {
        "dev": "dotenvx run --env-file=.env --env='HELLO=World' --overload expo start"
    }
}
```
to this:
```json
{
    "scripts": {
        "dev": "pnpm env:dev && expo start",
        "env:dev": "dotenvx run --env-file=.env.development --env='HELLO=World' --overload"
    }
}
```

This is just an idea and I don't know if something like this would even be possible.
Else this would have to be solved in some kind of script to start the dev server, which is an option but not the best DX as it adds complexity to the setup.

**Comment by motdotla** [comment]
this is available on `v0.20.0` and greater. (currently just for traditional non-encrypted runs). (adding to encrypted .env.vault runs as well shortly)

**Comment by motdotla** [comment]
@feledori want to give it a shot by installing `v0.20.0` and see if the functionality works as well as `cross-env`?

**Comment by feledori** [comment]
> @feledori want to give it a shot by installing `v0.20.0` and see if the functionality works as well as `cross-env`?

Working as expected on mac os! Trying later on windows.

**Comment by feledori** [comment]
> could you modify to this?
> 
> ```
> {
>     "scripts": {
>         "dev": "pnpm env:dev expo start",
>         "env:dev": "dotenvx run --env-file=.env.development --env='HELLO=World' --overload --"
>     }
> }
> ```

Tried doing this and its working as expected on macos. But windows on the other hand throws an error:
![image](https://github.com/dotenvx/dotenvx/assets/18011004/749d84f2-e0bb-4761-9b46-81ce20439628)

It says in german:
The "which" command is either misspelled or could not be found.

**Comment by motdotla** [comment]
if you downgrade to `0.20.0` does it work? i think the windows bug was introduced in `0.20.1` (which I'll work on a fix for asap)

**Comment by jpnws** [comment]
I'm on Windows 11, and I feel like the issue mentioned above:

https://github.com/dotenvx/dotenvx/issues/92#issuecomment-1964412187

looks a bit familiar to what I've encountered today.

I'm running an Express app and my `package.json` has the following script:

`"start:dev": "dotenvx run -- nodemon ./src/server.js"`

And, I'm getting the following error:

```
uname@usys MINGW64 ~/Projects/express1 (main)
$ npm run start:dev

> express1@1.0.0 start:dev
> dotenvx run -- nodemon ./src/server.js

[dotenvx@0.20.2] injecting env (8) from .env
The system cannot find the path specified.
Command failed with exit code 1: /c/Users/uname/AppData/Roaming/npm/nodemon ./src/server.js
command [nodemon ./src/server.js] failed

  try without dotenvx: [nodemon ./src/server.js]

if that succeeds, then dotenvx is the culprit. report issue:
<https://github.com/dotenvx/dotenvx/issues/new>
```

Is it correct that I might be running into a similar issue?

If not I can file a new GitHub issue if one doesn't already exist for this type of issue.

Is there any information you might need if I'm missing some context?

**Comment by motdotla** [comment]
yes, almost certainly. I just released `v0.21.0` which should fix this for windows users. can you try updating to it?

**Comment by jpnws** [comment]
> yes, almost certainly. I just released `v0.21.0` which should fix this for windows users. can you try updating to it? @jpnws

Excellent! I've tried `v0.21.0` just now and I no longer see the error.
